stdlib/bytes.gr

```diff
@@ -0,0 +1,466 @@
+import Memory from "runtime/unsafe/memory"
+import WasmI32 from "runtime/unsafe/wasmi32"
+import WasmI64 from "runtime/unsafe/wasmi64"
+import WasmF32 from "runtime/unsafe/wasmf32"
+import WasmF64 from "runtime/unsafe/wasmf64"
+import Conv from "runtime/unsafe/conv"
+import { tagSimpleNumber, allocateBytes, allocateString } from "runtime/dataStructures"
+import Exception from "runtime/exception"
+import Int32 from "int32"
+
+let _SIZE_OFFSET = 4n;
+let _VALUE_OFFSET = 8n;
+let _INT8_BYTE_SIZE = 1n;
+let _INT16_BYTE_SIZE = 2n;
+let _INT32_BYTE_SIZE = 4n;
+let _FLOAT32_BYTE_SIZE = 4n;
+let _INT64_BYTE_SIZE = 8n;
+let _FLOAT64_BYTE_SIZE = 8n;
+
+/** Converts Number to WasmI32 */
+@disableGC
+let numberToWasmI32 = n => {
+  let m = WasmI32.fromGrain(n)
+  if (WasmI32.eqz(WasmI32.and(m, 1n))) {
+    let i32 = Int32.fromNumber(n)
+    let res = Conv.fromInt32(i32)
+    Memory.free(WasmI32.fromGrain(i32))
+    res
+  } else {
+    // fast path
+    WasmI32.shrS(m, 1n)
+  }
+}
+
+/** Throws an exception if the index specified is out-of-bounds */
+@disableGC
+let ensureIndexIsInBounds = (i, byteSize, max) => {
```

Comment:
Do we have a preference over `ensure` and `assert`?

Reply:
I don't quite like `assert`, but I could go for something like `checkIndexIsInBounds`.

Reply:
I'd prefer `check` I think.